test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test "creates valid reply" do
+    login_as(@admin)
+    post :reply, params: { message: { subject: "Valid Message", :body => "message with subject" }, id: @message.id }
+
+    MessageWorker.drain
+    msg = Message.last
+    assert "sent", msg.state
+    assert_equal "message with subject", msg.body
+  end
+
+  test "not creates invalid message" do
+    login_as(@admin)
+    post :reply, params: { message: { subject: nil, body: "message with nil subject" }, id: @message.id }
+
+    MessageWorker.drain
+    msg = Message.last
+    assert_not_equal "message with nil subject", msg.body
```

Comment:
also `assert "unsent", msg.state` here?

Reply:
unsent is the initial state, and if validations fails we don't create object in db  so we check that there is  not last message with our subject which we have sent recently


Messages can be in 1 of 3 states:
# * +unsent+ - The message has not yet been sent.  This is the *initial* state.
# * +queued+ - The message has been queued for future delivery.
# * +sent+ - The message has been sent.